common/fee/utils.go

```diff
@@ -1,23 +1,55 @@
 package fee
 
 import (
+	"fmt"
 	"math/big"
 
+	"github.com/smartcontractkit/chainlink/v2/common/fee/types"
 	bigmath "github.com/smartcontractkit/chainlink/v2/core/utils/big_math"
 
 	"github.com/shopspring/decimal"
 )
 
-func GetMaxFeePrice(userSpecifiedMax, maxFee *big.Int) *big.Int {
-	return bigmath.Min(userSpecifiedMax, maxFee)
-}
-
-func CapFeePrice(calculatedFeePrice, userSpecifiedMax, maxFeePriceWei *big.Int, feeLimit uint32, multiplier float32) (maxFeePrice *big.Int, chainSpecificFeeLimit uint32) {
+// Returns the minimum of the calculated fee price, user specified max fee price and maxFeePrice for the node,
+// and the adjusted chain specific fee limit.
+func CapFeePrice(calculatedFeePrice, userSpecifiedMax, maxFeePriceAllowed *big.Int, feeLimit uint32, multiplier float32) (maxFeePrice *big.Int, chainSpecificFeeLimit uint32) {
 	chainSpecificFeeLimit = ApplyMultiplier(feeLimit, multiplier)
-	maxFeePrice = GetMaxFeePrice(userSpecifiedMax, maxFeePriceWei)
+	maxFeePrice = FeePriceLimit(userSpecifiedMax, maxFeePriceAllowed)
 	return bigmath.Min(calculatedFeePrice, maxFeePrice), chainSpecificFeeLimit
 }
 
+func FeePriceLimit(userSpecifiedMax, maxFee *big.Int) *big.Int {
+	return bigmath.Min(userSpecifiedMax, maxFee)
```

Comment:
If `FeePriceLimit` is just doing `min`, what value does the helper func add? Isn't that just more difficult to read?

Reply:
thought that it added a layer of abstraction, but i agree with you. It makes the code more readable. Thanks for bringing that up, not sure why we had that function initially but lets get rid of that function.